components/feature/session/src/main/java/mozilla/components/feature/session/SessionProvider.kt

```diff
@@ -70,7 +71,6 @@ class SessionProvider(
      * Stops this provider and periodic saves.
      */
     fun stop() {
-        sessions.clear()
```

Comment:
We shouldn't clear the sessions on stop. We call stop when the activity is paused, but when we resume we're reusing the same instance of GeckoView, but then we no longer have its engine session. It also doesn't want us to set a new session one as long as the old one is still open. So, best to hold to the sessions. For tabbing, we're gonna have to start cleaning up session anyway.